test/cmdlineTests/gas_test_abiv2/output

```diff
@@ -2,7 +2,7 @@
 ======= gas_test_abiv2/input.sol:C =======
 Gas estimation:
 construction:
-   1140 + 1096600 = 1097740
```

Comment:
That's quite an increase...

Reply:
It's only without the yul optimizer turned on, though - do we care about that?

Reply:
If we want to keep this down in the un-optimized case, I'd probably need to add some more case distinctions - so far the idea was to rely on the optimizer for that and the gas test case with the optimizer turned doesn't change.

Reply:
Can you explain where the increase comes from? There are no calldata variables here, are there?

Reply:
My guess would be the local ``baseRef`` variable and ``<elementValues>`` variables in ``abiEncodingFunctionSimpleArray``. The yul optimizer will eliminate them in the non-calldata cases, but I'm not sure how the old optimizer is able to handle them.

Reply:
ah right!